unittests/CppInterOp/DynamicLibraryManagerTest.cpp

```diff
@@ -2,55 +2,216 @@
+  EXPECT_TRUE(Cpp::GetFunctionAddress(MangleNameForDlsym("ret_2").c_str()));
+}
+
+TEST(DynamicLibraryManagerTest, LibrariesAutoloadExtraCoverage) {
+  EXPECT_TRUE(Cpp::CreateInterpreter());
+
+  // Libraries Search path by default is set to main executable directory.
+  std::string BinaryPath = GetExecutablePath(nullptr, nullptr);
+  llvm::StringRef Dir = llvm::sys::path::parent_path(BinaryPath);
+  Cpp::AddSearchPath(Dir.str().c_str());
+
+  // Force ExecutionEngine to be created.
+  Cpp::Process("");
+
+  // Autoload turn ON
+  Cpp::SetLibrariesAutoload(true);
+  // Check autorum status (must be turned ON)
+  EXPECT_TRUE(Cpp::GetLibrariesAutoload());
+
+  // Cover: MU getName()
+  std::string res;
+  llvm::raw_string_ostream rss(res);
+  const llvm::orc::SymbolNameVector syms;
+  Cpp::AutoLoadLibrarySearchGenerator::AutoloadLibraryMU MU("test", syms);
+  rss << MU;
+  EXPECT_STRNE("", rss.str().c_str()) << "MU problem!";
+
+  // Cover: LoadLibrary error
+  // if (DLM->loadLibrary(lib, false) != DynamicLibraryManager::LoadLibResult::kLoadLibSuccess) {
+  //   LLVM_DEBUG(dbgs() << "MU: Failed to load library " << lib);
+  //   string err = "MU: Failed to load library! " + lib;
+  //   perror(err.c_str());
+  // } else {
+  // Find library with "ret_value" symbol defined and exported
+  std::string PathToTestSharedLib3 =
+      Cpp::SearchLibrariesForSymbol(MangleNameForDlsym("ret_val").c_str(), /*system_search=*/false);
+  // If result is "" then we cannot find this library.
+  EXPECT_STRNE("", PathToTestSharedLib3.c_str())
+      << "Cannot find: '" << PathToTestSharedLib3 << "' in '" << Dir.str() << "'";
+  // Remove library for simulate load error
+  llvm::sys::fs::remove(PathToTestSharedLib3, true);
+  EXPECT_TRUE(Cpp::GetLibrariesAutoload());
+  // FIXME: Conda returns false to run this code on osx.
+  EXPECT_FALSE(Cpp::GetFunctionAddress(MangleNameForDlsym("ret_val").c_str()));
+
+  // Cover
+  // } else {
+  //   // Collect all failing symbols, delegate their responsibility and then
+  //   // fail their materialization. R->defineNonExistent() sounds like it
+  //   // should do that, but it's not implemented?!
+  //   failedSymbols.insert(symbol);
+  // TODO: implement test this case
+
+  // Cover
+  // if (!failedSymbols.empty()) {
+  //   auto failingMR = R->delegate(failedSymbols);
+  //   if (failingMR) {
+  //     (*failingMR)->failMaterialization();
+  // TODO: implement test this case
+
+  // Cover
+  // void discard(const llvm::orc::JITDylib &JD, const llvm::orc::SymbolStringPtr &Name) override {}
+  // TODO: implement test this case
+
+  // Cover
+  // if (Path.empty()) {
+  //   LLVM_DEBUG(dbgs() << "DynamicLibraryManager::lookupLibMaybeAddExt(): "
+  // TODO: implement test this case
+
+  // Cover
+  // platform::DLClose(dyLibHandle, &errMsg);
+  // if (!errMsg.empty()) {
+  //   LLVM_DEBUG(dbgs() << "DynamicLibraryManager::unloadLibrary(): "
+  // TODO: implement test this case
```

Comment:
We should write these tests, right?

Reply:
These are quite unexpected cases, even some of them are quite difficult to simulate even if you want to, but it's really good to implement these cases for better test coverage.

Reply:
Yes, if we are not planning to implement these tests as part of this PR we should probably remove the code and keep track of this in a github issue.

Reply:
They are planned unless we rush with the merge...

Reply:
Well, there is no rush but this is a blocker for xeus-cpp. If we can implement them within a week or so then that's probably good. Otherwise we can make a subsequent PR.